### ISSUE TITLE
I3MCTree missing in hdf5 and root files (Trac #1082)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1076]
When I write i3-files into hdf5 and/or root files I3MCTree is missing. The files are only copied, there are no cuts or other processing involved and everything is booked (BookEverything = True). This issue doesn't occur with icerec release V04-06-00, but with V04-11-02, so it changed somewhere in between.

Migrated from https://code.icecube.wisc.edu/ticket/1082
```json
{
    "status": "closed", 
    "changetime": "2015-08-10T20:31:17", 
    "description": "When I write i3-files into hdf5 and/or root files I3MCTree is missing. The files are only copied, there are no cuts or other processing involved and everything is booked (BookEverything = True). This issue doesn't occur with icerec release V04-06-00, but with V04-11-02, so it changed somewhere in between.", 
    "reporter": "lisa", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1439238677802127", 
    "component": "combo reconstruction", 
    "summary": "I3MCTree missing in hdf5 and root files", 
    "priority": "blocker", 
    "keywords": "I3MCTree, I3TableWriter", 
    "time": "2015-08-03T12:29:20", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150803T16:57:31: 
* richman changed status from "new" to "closed"
* richman commented:

> Recently, a new version of I3MCTree was rolled out.  The new I3MCTree breaks the old I3MCTree tableio converter.  That converter was considered mostly useless.  Usually you just want the "primary", which you can pluck from the I3MCTree with frame[‘MCPrimary’] = frame[‘I3MCTree’].primaries[0], or with weighting.get_weighted_primary() if you want something that’s robust against coincident showers.


* richman changed resolution from "" to "wontfix"

**Comment by icecube-trac** [comment]
Trac update at 20150808T16:00:17: 
* koskinen changed status from "closed" to "reopened"
* koskinen changed resolution from "wontfix" to ""
* koskinen commented:

> A parochial view of what most people want does not in this instance include work done by PINGU and other lower energy groups. The ability to write out the full particle tree is an essential aspect of MC validation in multiple WGs and needs to be re-introduced for I3MCTree.

**Comment by icecube-trac** [comment]
Trac update at 20150809T15:49:02: richman commented:

> That's a fair point.  This has come up several times since the new I3MCTree rolled out (and even before), and I've gotten the impression that the usual suspects are mostly uninterested in re-enabling the old behavior.  The above "workaround" does seem to suffice for most use cases, but if someone really needs this and someone knows how to fix it, then it should happen.  Sorry for the noise.

**Comment by icecube-trac** [comment]
Trac update at 20150809T17:35:04: 
* olivas changed status from "reopened" to "accepted"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20150809T17:35:36: 
* olivas changed milestone from "" to "offline V15-07-00"
* olivas changed priority from "major" to "blocker"

**Comment by icecube-trac** [comment]
Trac update at 20150809T20:09:39: 
* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "accepted" to "closed"
* jvansanten commented:

> In changeset 136020:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="136020"
> Bring back I3MCTreeConverter. Fixes #1076.
> 
> Along the way, fix a bug that would have prevented I3MCTreeConverter from
> picking up instances of I3LinearizedMCTree.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20150810T20:31:17: 
* olivas changed milestone from "offline V15-07-00" to ""
* olivas commented:

> Milestone offline V15-07-00 deleted
